### PR TITLE
Support Error Publishing for Websocket Frame Errors

### PR DESCRIPTION
**Pull request by msm1992**
Fixes https://github.com/wso2/product-apim/issues/12197

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java [hunk 1]
```diff
@@ -215,6 +250,20 @@ private void addThrottledErrorPropertiesToChannel(ChannelHandlerContext ctx) {
         WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE, "Message Throttled Out");
     }
 
+    private void addAuthErrorPropertiesToChannel(ChannelHandlerContext ctx) {
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_CODE,
+                WebSocketApiConstants.FrameErrorConstants.API_AUTH_GENERAL_ERROR);
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java:255]: @msm1992 Can we set the actual error code in the InboundProcessorResponseDTO ?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java [hunk 2]
```diff
@@ -215,6 +250,20 @@ private void addThrottledErrorPropertiesToChannel(ChannelHandlerContext ctx) {
...
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE,
+                "Authentication Failure");
+    }
+
+    private void addOtherErrorPropertiesToChannel(ChannelHandlerContext ctx) {
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_CODE,
+                WebSocketApiConstants.FrameErrorConstants.OTHER_ERROR);
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE,
+                "Unclassified Error");
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java:264]: same here. why cant we publish the error code and message in the InboundProcessorResponseDTO itself?

**Comment by msm1992** [comment]
Closing this PR as updates were sent in https://github.com/wso2/carbon-apimgt/pull/11187

**Pull request closed without merging**